src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -464,64 +306,82 @@ private TestExecutionResult testPluginAgainst(
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
 
-                if (localCheckoutProvided()) {
-                    if (!onlyOnePluginIncluded()) {
-                        File localCheckoutPluginDir =
-                                new File(config.getLocalCheckoutDir(), plugin.name);
-                        File pomLocalCheckoutPluginDir =
-                                new File(localCheckoutPluginDir, "pom.xml");
-                        if (pomLocalCheckoutPluginDir.exists()) {
-                            LOGGER.log(
-                                    Level.INFO,
-                                    "Copying plugin directory from {0}",
-                                    localCheckoutPluginDir.getAbsolutePath());
+            if (localCheckoutProvided()) {
+                if (!onlyOnePluginIncluded()) {
+                    File localCheckoutPluginDir =
+                            new File(config.getLocalCheckoutDir(), plugin.name);
+                    File pomLocalCheckoutPluginDir = new File(localCheckoutPluginDir, "pom.xml");
+                    if (pomLocalCheckoutPluginDir.exists()) {
+                        LOGGER.log(
+                                Level.INFO,
+                                "Copying plugin directory from {0}",
+                                localCheckoutPluginDir.getAbsolutePath());
+                        try {
                             org.codehaus.plexus.util.FileUtils.copyDirectoryStructure(
                                     localCheckoutPluginDir, pluginCheckoutDir);
-                        } else {
-                            cloneFromSCM(
-                                    pomData, plugin.name, plugin.version, pluginCheckoutDir, "");
+                        } catch (IOException e) {
+                            throw new UncheckedIOException(e);
                         }
                     } else {
-                        // TODO this fails when it encounters symlinks (e.g.
-                        // work/jobs/…/builds/lastUnstableBuild), and even up-to-date versions of
-                        // org.apache.commons.io.FileUtils seem to not handle links, so may need to
-                        // use something like
-                        // http://docs.oracle.com/javase/tutorial/displayCode.html?code=http://docs.oracle.com/javase/tutorial/essential/io/examples/Copy.java
-                        LOGGER.log(
-                                Level.INFO,
-                                "Copy plugin directory from {0}",
-                                config.getLocalCheckoutDir().getAbsolutePath());
+                        cloneFromScm(
+                                pomData.getConnectionUrl(),
+                                config.getFallbackGitHubOrganization(),
+                                getScmTag(pomData, plugin.name, plugin.version),
+                                pluginCheckoutDir);
```

Comment:
While I was here I deduplicated some logic that was duplicated in the various callers of `cloneFromSCM`.